server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -283,6 +281,43 @@ public boolean setExpiration(String name, Instant expiration) {
     return Optional.empty();
   }
 
+  /**
+   * @param names of secrets series to look up secrets by.
+   * @return Secrets matching input parameters.
+   */
+  public List<SecretSeriesAndContent> getSecretsByName(List<String> names) {
+    checkArgument(!names.isEmpty());
+
+    SecretContentDAO secretContentDAO = secretContentDAOFactory.using(dslContext.configuration());
+    SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
+
+    List<SecretSeries> multipleseries = secretSeriesDAO.getSecretsSeriesByName(names);
+
+
+    Iterator siter = multipleseries.iterator();
+
+    List<SecretSeriesAndContent> ret = new ArrayList<SecretSeriesAndContent>();
+
+    while (siter.hasNext()) {
+
+      SecretSeries series = (SecretSeries) siter.next();
+
+      if (series.currentVersion().isPresent()) {
+        long secretContentId = series.currentVersion().get();
+        Optional<SecretContent> secretContent =
+                secretContentDAO.getSecretContentById(secretContentId);
+        if (secretContent.isPresent()) {
+          ret.add(SecretSeriesAndContent.of(series, secretContent.get()));
+        } else {
+          // TODO: how to handle this?
```

Comment:
Since `getSecretsSeriesByName` requires `SECRETS.CURRENT.IsNotNull()`, this should be impossible, so it's probably fine to throw an exception here.